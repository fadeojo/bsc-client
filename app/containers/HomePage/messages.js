/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  login: {
    id: 'app.components.HomePage.header',
    defaultMessage: 'Login',
  },
});
