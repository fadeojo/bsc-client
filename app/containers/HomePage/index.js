/*
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a necessity for you then you can refactor it and remove
 * the linting exception.
 */

import React from 'react';
import { Link } from 'react-router-dom';
import { object } from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { Button } from 'antd';
import messages from './messages';
// import { WaterWave } from 'ant-design-pro/lib/Charts';
import './HomePage.scss';
import healthImage from '../../images/healthcare-sign.svg';

// eslint-disable-next-line
export default class HomePage extends React.PureComponent {
  render() {
    return (
      <div className="home-component">
        <div
          style={{
            display: 'flex',
            flexDirection: 'row',
            height: '100%',
            justifyContent: 'center',
          }}
        >
          <div
            style={{
              display: 'flex',
              justifyContent: 'center',
              flexDirection: 'column',
            }}
          >
            <img src={healthImage} alt="" />
            <Button type="primary" style={{ width: 225, height: 42 }}>
              <Link to="/login">
                <FormattedMessage {...messages.login} />
              </Link>
            </Button>
          </div>
        </div>
      </div>
    );
  }
}

HomePage.propTypes = {
  history: object.isRequired,
};
