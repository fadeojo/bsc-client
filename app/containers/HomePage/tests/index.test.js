import React from 'react';
import renderer from 'react-test-renderer';
import { FormattedMessage, IntlProvider } from 'react-intl';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import HomePage from '../index';
import messages from '../messages';

configure({ adapter: new Adapter() });

describe('<HomePage />', () => {
  it('should render correctly', () => {
    const renderedComponent = renderer.create(
      <IntlProvider locale={'en'} key={'en'} messages={messages}>
        <HomePage />
      </IntlProvider>
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render the page message', () => {
    const renderedComponent = shallow(<HomePage />);
    expect(
      renderedComponent.contains(<FormattedMessage {...messages.login} />)
    ).toEqual(true);
  });
});
