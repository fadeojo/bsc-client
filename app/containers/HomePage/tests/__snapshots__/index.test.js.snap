// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render correctly 1`] = `
<div
  className="home-component"
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <img
        alt=""
        src="IMAGE_MOCK"
      />
      <button
        className="ant-btn ant-btn-primary"
        onClick={[Function]}
        style={
          Object {
            "height": 42,
            "width": 225,
          }
        }
        type="button"
      >
        <span>
          Login
        </span>
      </button>
    </div>
  </div>
</div>
`;
